system/Database/MySQLi/Connection.php

```diff
@@ -368,11 +368,17 @@ public function escapeLikeStringDirect($str)
     /**
      * Generates the SQL for listing tables in a platform-dependent manner.
      * Uses escapeLikeStringDirect().
+     *
+     * @param false|string $tableName
      */
-    protected function _listTables(bool $prefixLimit = false): string
+    protected function _listTables(bool $prefixLimit = false, $tableName = false): string
     {
         $sql = 'SHOW TABLES FROM ' . $this->escapeIdentifiers($this->database);
 
+        if ($tableName !== false) {
+            return $sql . " LIKE '" . $tableName . "'";
```

Comment:
SQL injection can be done. Use the escape function for it.

Reply:
By the way, why LIKE?

Reply:
Its built in to SHOW TABLES. You can use WHERE but the field name is dynamic and changes. LIKE without wildcard should suffice just fine.

These two queries are equivalent.

```SQL
SHOW TABLES FROM `newdb` LIKE 'db_user'

SHOW TABLES FROM `newdb` WHERE Tables_in_newdb = 'db_user'
```